File: Models/Task.swift
//
//  Task.swift
//  HocusFocusApp
//
//  Created by liz cruz
//  Student ID: 101470163

import Foundation

struct Task: Identifiable, Codable {
    var id: String
    var title: String
    var dueDate: String
    var isCompleted: Bool
    var projectId: String

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case title
        case dueDate = "dueDate"
        case isCompleted = "isCompleted"
        case projectId = "projectId"
    }
}

File: HocusFocusBackend/models/Task.js
//101470163 Liz Cruz

const mongoose = require("mongoose");

const TaskSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true
  },
  dueDate: {
    type: Date,
    required: true
  },
  isCompleted: {
    type: Boolean,
    default: false
  },
  projectId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Project",
    required: true
  }
});

module.exports = mongoose.model("Task", TaskSchema);
